packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -151,10 +169,19 @@ export class Stack extends Construct implements ITaggable {
 
   /**
    * Returns the parent stack if this stack is nested.
+   * This is `undefined` for top-level (non-nested) stacks.
    *
    * @experimental
    */
-  public readonly parentStack?: Stack;
+  public readonly nestedStackParent?: Stack;
```

Comment:
`protected` for both?

Also, doesn't it make more sense to have a single property `nestedStackInfo?: NestedStackInfo` ?

Reply:
I've changed this slightly so `NestedStack` only needs to set `nestedStackResource` and the stack is deduced from it (`Stack.of`). It can't be protected since it is used by thing like the assertion library and the `addDependency` code.

As mentioned, I think this will undergo some more changes when we move `NestedStack` into `@aws-cdk/core`.